tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+
+@pytest.mark.sanity
+@pytest.mark.bridge
+@pytest.mark.nat
+@pytest.mark.vlan
+@pytest.mark.firmware
+class TestFirmware(object):
+
+    @pytest.mark.firmware_create
+    def test_firmware_create(self, upload_firmware, instantiate_testrail, instantiate_project, test_cases):
+        if upload_firmware != 0:
+            instantiate_testrail.update_testrail(case_id=test_cases["create_fw"], run_id=instantiate_project,
```

Comment:
How does this work when testrail is disabled?

Reply:
so i am having a dummy class that has same methods name, so if we want to disable testrail, then that class instance is passed to all testcase and that function does nothing